draft-ietf-quic-transport.md

```diff
@@ -1890,6 +1897,46 @@ different static key or server identifier.
 Note that Stateless Reset messages do not have any cryptographic protection.
 
 
+#### Connection IDs and Stateless Reset {#reset-cid}
+
+This design relies on the client sending a connection ID in all packets so that
+the server can use the connection ID from a packet to reset the connection.  A
+server that uses this design cannot allow clients to omit a connection ID.  That
+is, it cannot set the truncate_connection_id transport parameter
+({{transport-parameter-definitions}}) to 0.
+
+Ensuring that the server sends the correct connection ID in its stateless reset
+presents an additional challenge.  Truncation of the connection ID in packets
+that use the short header means that the entire connection ID might not be
+present in the packet that triggers the stateless reset.  Furthermore, the
+server cannot know how many octets from the connection ID that the client
+requested be included in the short header.
+
+A server is expected to have a minimum number of octets that are needed for
+correct routing.  Thus, the server can always know that at least this many
+octets of the packet it receives contain a connection ID.
+
+If the server uses a longer connection ID than the client, there is the
+potential for the additional octets to be used to reveal that the stateless
+reset is not a regular packet.  This is not a serious issue, as it is already
+possible to use other fields to distinguish them from other packet.
+
+If a client requires more octets of the connection ID for routing than the
+server requires, those octets will not appear in the packet that the server
+receives.  In this case, the stateless reset might not reach the client.
+
+A server can avoid this by ensuring that it always receives a connection ID that
+is equal to or longer than the connection ID requested by the client.  In
+choosing a connection ID, the server already knows the client's preferred
+transaction ID length, and so can request that the client provide a similarly
+long connection ID.
```

Comment:
This is probably the right choice, but again departs from our current semantics that servers don't have to look at the client's transport params before generating their own.

Reply:
Yeah, but this is still a degree shy of making it a negotiation.